src/sparse/KokkosSparse_sptrsv_supernode.hpp

```diff
@@ -1300,87 +1419,47 @@ invert_supernodal_columns(KernelHandle kernelHandle, bool unit_diag, int nsuper,
         #ifdef KOKKOS_SPTRSV_SUPERNODE_PROFILE
         timer.reset ();
         #endif
-        KokkosBlas::trmm (&side_char, &uplo_char,
-                          &tran_char, &diag_char,
-                          one, Ljj, Lij);
+        if(run_trmm_on_device) {
+          Kokkos::View<scalar_t**, Kokkos::LayoutLeft, trmm_memory_space, Kokkos::MemoryUnmanaged>
+            devL (trmm_dwork.data(), nsrow, nscol);
+          auto devLjj = Kokkos::subview (devL, range_type (0, nscol), Kokkos::ALL ());
+          auto devLij = Kokkos::subview (devL, range_type (nscol, nsrow), Kokkos::ALL ());
+
+          Kokkos::deep_copy(devL, viewL);
```

Comment:
fence to ensure deep_copy completes before reading from devL

Reply:
Thank you, @e10harvey. Just to make sure, do we really need these fences after and before deep-copies?

Reply:
Can you confirm the unit-tests pass when they exercise this new code path with the deep copies? Particularly when CudaUVM is used? Please ensure the unit-tests exercise this code path even if that means the unit tests must set a tolerance to a non-default value when the handle is created.